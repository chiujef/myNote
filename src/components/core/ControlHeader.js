import PropTypes from 'prop-types';
import React from 'react';
import {StyleSheet, View} from 'react-native';

const ControlHeader = ({children}) => {
  return <View style={styles.container}>{children}</View>;
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    top: 0,
    width: '100%',
    height: 50,
    borderTopColor: '#b3bec9',
    borderTopWidth: 1,
    flexDirection: 'row',
    backgroundColor: '#F5FCFF',
  },
});

ControlHeader.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ControlHeader;
