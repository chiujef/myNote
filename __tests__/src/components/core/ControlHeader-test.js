/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import {render} from 'react-native-testing-library';

import ControlHeader from '@components/core/ControlHeader';

describe('ControlHeader', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer
        .create(
          <ControlHeader>
            <Text>test</Text>
          </ControlHeader>,
        )
        .toJSON();

      expect(component).toMatchSnapshot();
    });

    it('should render the child component', () => {
      const {queryByType} = render(
        <ControlHeader>
          <Text>test</Text>
        </ControlHeader>,
      );

      expect(queryByType('Text')).not.toBeNull();
    });
  });
});
