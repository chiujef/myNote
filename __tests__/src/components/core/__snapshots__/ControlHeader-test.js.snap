// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlHeader Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "borderTopColor": "#b3bec9",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <Text>
    test
  </Text>
</View>
`;
